reactor-core/src/main/java/reactor/core/publisher/FluxDistinctUntilChanged.java

```diff
@@ -103,8 +111,7 @@ public boolean tryOnNext(T t) {
 				return true;
 			}
 
-			if (Objects.equals(lastKey, k)) {
-				lastKey = k;
+			if (null != lastKey && keyComparator.test(lastKey, k)) {
 				return false;
```

Comment:
`lastKey` should still be initialized. Also this block would need to be protected against the keyComparator throwing an exception, like what is done just above.